### DISCUSSION TITLE
commercial use?

### DISCUSSION THREAD

**Discussion by radopas** [javamelody/javamelody#1163] [General]
Hello,

Could you please clarify is it allowed to use in commercial applications?
We are a vendor that develops a system with a web module and Javamelody is integrated with the web module. My understanding is that Javamelody depends on JRobin, which is under LGPL license and this depreciates its use in our case.

**Comment by evernat** [comment]
You can use javamelody in commercial applications indeed. I have done and still do that in commercial applications in my daily job countless times.

[javamelody license is Apache-2.0.](https://github.com/javamelody/javamelody/blob/master/LICENSE)

And as you have seen, javamelody has a dependency on [JRobin, which has a LGPL 2.1 license](https://sourceforge.net/p/jrobin/code/ci/master/tree/LICENSE.txt). Note that LGPL license is very different than GPL license, in particular for commercial use and in particular if you don't modify the JRobin library.

Quote of [wikipedia on LGPL](https://en.wikipedia.org/wiki/GNU_Lesser_General_Public_License) : 
> The license allows developers and companies to use and integrate a software component released under the LGPL into their own (even [proprietary](https://en.wikipedia.org/wiki/Proprietary_software)) software without being required by the terms of a strong [copyleft](https://en.wikipedia.org/wiki/Copyleft) license to release the source code of their own components. However, any developer who modifies an LGPL-covered component is required to make their modified version available under the same LGPL license.
[...]
Paragraph 5 of the LGPL version 2.1 states:
A program that contains no derivative of any portion of the Library, but is designed to work with the Library by being compiled or linked with it, is called a "work that uses the Library". Such a work, in isolation, is not a derivative work of the Library, and therefore falls outside the scope of this License.

In short, using Maven or anything similar to compile and link your application to javamelody (and to jrobin) does not make your application a derivative work of jrobin and does not restrict your application's license. If you want to modify the classes in the jrobin library, then you would need to make available those changes under LGPL (like the [rrd4j](https://github.com/rrd4j/rrd4j) fork did, even if adding new code with Apache license is debatable in that fork).

Anyway, if you can't use a LGPL dependency in your case, that's probably more restrictive than what I have ever seen and I am sorry for you.
